File: DKS-SPA/src/app/core/_models/s-dev-dtr-fgt-result-report.ts
import { utilityConfig } from "../utility/utility-config";
import { Pagination } from "./pagination";

export class SDevDtrFgtResultReport extends Pagination {
  reportType: string;     //determine Dev or Buy plan
  
  devSeason: string;      //DEV use
  buyPlanSeason: string;  //DHO use
  factory:string;         //DHO use

  article: string;
  cwaDateS: string;
  cwaDateE: string;
  modelNo: string;
  modelName: string;
  

  constructor() {
    super();

    this.devSeason = "";

    this.buyPlanSeason = "";
    this.factory = ""; //預設翔鴻程

    this.reportType = "";
    this.article = "";
    this.cwaDateS = "";
    this.cwaDateE = "";
    this.modelNo = "";
    this.modelName = "";

    this.isPaging = true; //開分頁

  }
  
  public setPagination(pagination: Pagination) {

    this.currentPage = pagination.currentPage;
    this.itemsPerPage = pagination.itemsPerPage;
    this.totalItems = pagination.totalItems;
    this.totalPages = pagination.totalPages;
  }
}

File: DKS-API/DTOs/SDevDtrFgtResultReport.cs
using System;
using DKS_API.Helpers;

namespace DKS_API.DTOs
{
    public class SDevDtrFgtResultReport : PaginationParams
    {
        public string devSeason { get; set; }

        public string buyPlanSeason { get; set; }
        public string factory { get; set; }

        public string reportType { get; set; }
        public string article { get; set; }
        public string cwaDateS { get; set; }
        public string cwaDateE { get; set; }
        public string modelNo { get; set; }
        public string modelName { get; set; }

    }
}